README.md

```diff
@@ -0,0 +1,235 @@
+
+However the hybrid KEM Kyber768X25519 is constructed using a security
+preserving KEM combiner and a NIKE to KEM adapter with semantic
+security so that the resulting hybrid KEM is IND-CCA2 in QROM.
+
+Here's a diagram of the pqXX pattern which we use:
+
+```mermaid
+sequenceDiagram
+    Client-)Server: e
+    Server-)Client: ekem, s
+    Client-)Server: skem, s
+    Server-)Client: skem
+```
+
+## Sphinx
+
+We use the Sphinx cryptographic packet format and allow it's geometry
+to be completely configurable to accomodate various networking
+requirements. Additionally the Sphinx can use any NIKE
+(non-interactive key exchange). We also developed a novel post quantum
+variation called KEM Sphinx. KEM (key encapsulation mechanism) Sphinx
+is twice as fast on the server side as the original NIKE Sphinx
+because it only requires one public key operation per hop instead of
+two. However it has the packet header overhead size penalty that grows
+linearly with the number of hops.
+
+And here are some Sphinx benchmarks using different KEMs and NIKEs:
+
+| Primitive | Sphinx type | nanoseconds/op | seconds/op |
+| :---      |  :---:      |     ---:       | ---:       |
+| X25519 | KEM | 80093 | 8.009×10−5 |
+| X25519 | NIKE | 160233 | 0.000160233 |
+| Kyber512 | KEM | 43758 | 4.3758e-5 |
+| Kyber768 | KEM | 57049 | 5.7049e-5 |
+| Kyber1024 | KEM | 72173 | 7.2173e-5 |
+| Kyber768 X25519 Hybrid | KEM | 87816 | 8.7816e-5 |
+| CTIDH512 | NIKE | 336995975 | 0.336995975 |
+| CTIDH1024 | NIKE | 18599579037 | 18.599579037 |
+| CTIDH2048 | NIKE | 17056742100 | 17.0567421 |
+| CTIDH1024 | KEM | 11408217346 | 11.408217346 |
+
+We can draw several conclusions from this table of benchmarks:
+
+1. KEM Sphinx is about twice as fast as NIKE Sphinx
+2. Kyber768 is faster than X25519
+3. Kyber768 X25519 Hybrid KEM Sphinx is almost as fast as X25519 NIKE
+   Sphinx but probably a lot more secure given that it's a post quantum
+   hybrid construction which still uses the classically secure X25519
+   NIKE.
+4. CTIDH is very slow and we probably don't want to use it for
+   Sphinx. We instead think it very useful for application level
+   encryption.
+
+Please also note that hybrid KEMs referred to above are constructed
+using a security preserving KEM combiner and a NIKE to KEM adapter
+with semantic security so that the resulting hybrid KEM is IND-CCA2 in
+QROM.
+
+## PKI/Directory Authority
+
+Mix network key management and distribution is handled by the
+directory authority system, a decentralized voting protocol that can
+tolerate (1/2 * n)-1 node outages.  Clients and mix nodes can talk to
+the dirauth system to get a published PKI document which is
+essentially a view of the network which contains public cryptographic
+keys and network connection information.
+
+# License
+
+GPLv3
+
+# Donations
+
+Your donations are welcomed and can be made through Open Collective [here.](https://opencollective.com/the-katzenpost-software-project)
```

Comment:
we shoudl make a few donation addresses for BTC, ZEC, etc, and an email address for donations of other things like servers, etc

Reply:
https://github.com/katzenpost/katzenpost/issues/358